src/dist/replication/meta_server/meta_service.cpp

```diff
@@ -732,28 +715,15 @@ void meta_service::on_start_recovery(configuration_recovery_rpc rpc)
 {
     configuration_recovery_response &response = rpc.response();
     ddebug("got start recovery request, start to do recovery");
-    int result = check_leader(rpc.dsn_request(), nullptr);
-    if (result == 0) // request has been forwarded to others
-    {
-        rpc.disable_auto_reply();
+    if (!check_status(rpc)) {
         return;
     }
 
-    if (result == -1) {
-        response.err = ERR_FORWARD_TO_OTHERS;
-    } else {
-        zauto_write_lock l(_meta_lock);
-        if (_started.load()) {
-            ddebug("service(%s) is already started, ignore the recovery request",
-                   dsn_primary_address().to_string());
-            response.err = ERR_SERVICE_ALREADY_RUNNING;
-        } else {
-            _state->on_start_recovery(rpc.request(), response);
-            if (response.err == dsn::ERR_OK) {
-                _recovering = false;
-                start_service();
-            }
-        }
+    zauto_write_lock l(_meta_lock);
```

Comment:
```suggestion
    zauto_write_lock l(_meta_lock);
    if (_started.load()) {
        ddebug("service(%s) is already started, ignore the recovery request",
               dsn_primary_address().to_string());
        response.err = ERR_SERVICE_ALREADY_RUNNING;
    } else {
```

Why you remove this part?

Reply:
yes, you are right. We can't refactor `on_start_recovery` by using `check_status`